### DISCUSSION TITLE
CSLA 6 prerelease 3 online

### DISCUSSION THREAD

**Discussion by rockfordlhotka** [MarimerLLC/csla#2679] [Announcements]
I am pleased to let everyone know that a new prerelease of CSLA 6 is now online.

You can find all the [changes in the backlog](https://github.com/MarimerLLC/csla/issues?q=is%3Aissue+is%3Aclosed+project%3AMarimerLLC%2Fcsla%2F11).

Most of the changes from prerelease 2 to 3 are feature enhancements requested by folks who've been testing CSLA 6 (**thank you so much!!**), along with fixing a couple bugs that have been discovered.

I continue to update the sample apps, which helps with testing (though not as much as _your_ feedback). Right now the following have been updated to .NET 6 and a recent CSLA 6 preview:

* BlazorExample
* MvcExample
* RazorPagesExample
* WinFormsExample
* WpfExample

Again, I can't stress enough how incredibly valuable it is when _you_ test on your existing codebases. Many of you are doing fantastic things with CSLA, and sometimes using it in ways I didn't anticipate. Your feedback on the substantial changes in CSLA 6 _now_ can help make everyone's life easier in the future!

**Comment by TheCakeMonster** [comment]
Oh wow, that's great! What a lovely New Year present, thanks!

I've managed to make it through the first pass of getting the Csla.Test project up and running again (which I've effectively replaced with a new project containing all of the existing tests.) It wasn't the most fun 3 days of my life, but it's been very productive; I can see from the failing tests that there are a few areas that are likely to cause people problems. Consider these known issues please folks!

1. Editable Lists don't seem to work well; they throw an exception when attempting to instantiate _deletedList using lazy loading. There is a chance the change that Rocky has made associated with lazy loading has already fixed this - or can be used to do so.
2. Anything involving object cloning (including any use of n-level undo) throws a NullReferenceException in SerializationFormatter.
3. It might be that inspectors are running more than once per request; there's something not quite right in that area it seems.
4. Use of the transactional behaviour offered by TransactionalDataPortal throws a NullReferenceException; this one I have already found and fixed, but it's not been submitted yet.

Once I've worked out how to upload the many hundreds of changes to the tests to GitHub without the world's lights dimming or GitHub's Actions runners being taken down by the many remaining test failures, I'll submit a PR so that the remaining issues can be more easily investigated.
